rmw_zenoh_cpp/src/rmw_zenoh.cpp

```diff
@@ -48,6 +48,102 @@
 #include "rmw/validate_namespace.h"
 #include "rmw/validate_node_name.h"
 
+namespace
+{
+
+//==============================================================================
+// A function to take ros topic names and convert them to valid Zenoh keys.
+// In particular, Zenoh keys cannot start or end with a /, so this function
+// will strip them out.
+// The Zenoh key is also prefixed with the ros_domain_id.
+// Performance note: at present, this function allocates a new string and copies
+// the old string into it. If this becomes a performance problem, we could consider
+// modifying the topic_name in place. But this means we need to be much more
+// careful about who owns the string.
+z_owned_keyexpr_t ros_topic_name_to_zenoh_key(
+  const char * const topic_name, size_t domain_id, rcutils_allocator_t * allocator)
+{
+  size_t start_offset = 0;
+  size_t topic_name_len = strlen(topic_name);
+  size_t end_offset = topic_name_len;
+
+  if (topic_name_len > 0) {
+    if (topic_name[0] == '/') {
+      // Strip the leading '/'
+      start_offset = 1;
+    }
+    if (topic_name[end_offset - 1] == '/') {
+      // Strip the trailing '/'
+      end_offset -= 1;
+    }
+  }
+
+  std::stringstream domain_ss;
+  domain_ss << domain_id;
+  char * stripped_topic_name = rcutils_strndup(
+    &topic_name[start_offset], end_offset - start_offset, *allocator);
+  z_owned_keyexpr_t keyexpr = z_keyexpr_join(
+    z_keyexpr(domain_ss.str().c_str()), z_keyexpr(stripped_topic_name));
+  allocator->deallocate(stripped_topic_name, allocator->state);
+
+  return keyexpr;
+}
+
+//==============================================================================
+const rosidl_message_type_support_t * find_type_support(
```

Comment:
I'm going to suggest we rename this to `find_msg_type_support`, just to be clear about what the function does.

Reply:
I've now done this in [9d9f6f6](https://github.com/ros2/rmw_zenoh/pull/86/commits/9d9f6f6597bf676cc7aaedeeb02fe23f63f43da4).